src/main/java/com/shopping_app/shoppingApp/service/UserService.java

```diff
@@ -0,0 +1,196 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.NotFoundException;
+import com.shopping_app.shoppingApp.Exceptions.UserAlreadyExist;
+import com.shopping_app.shoppingApp.config.JWT.JwtTokenProvider;
+import com.shopping_app.shoppingApp.domain.Address;
+import com.shopping_app.shoppingApp.domain.Cart;
+import com.shopping_app.shoppingApp.domain.User;
+import com.shopping_app.shoppingApp.model.Address.Response.AddressResponse;
+import com.shopping_app.shoppingApp.model.Enum.UserRole;
+import com.shopping_app.shoppingApp.model.User.Request.UserLoginRequest;
+import com.shopping_app.shoppingApp.model.User.Request.UserRegisterRequest;
+import com.shopping_app.shoppingApp.model.User.Request.UserUpdateRequest;
+import com.shopping_app.shoppingApp.model.User.Response.UserLoginResponse;
+import com.shopping_app.shoppingApp.model.User.Response.UserResponse;
+import com.shopping_app.shoppingApp.repository.AddressRepository;
+import com.shopping_app.shoppingApp.repository.CartRepository;
+import com.shopping_app.shoppingApp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final AddressRepository addressRepository;
+    private final CartRepository cartRepository;
+    private final PasswordEncoder bCryptPasswordEncoder;
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenProvider tokenProvider;
+
+    public UserResponse registerUser(UserRegisterRequest userRegisterRequest) {
+        Optional<User> isUserExist = userRepository.findByEmail(userRegisterRequest.getEmail());
```

Comment:
Same request on the renaming variable name here and other places in the file as applicable.

Reply:
done